test/e2e/common.go

```diff
@@ -82,3 +88,48 @@ func GetLatestReleaseOfMinor(ctx context.Context, minorRelease string) (string,
 	releaseMarker := fmt.Sprintf(latestReleaseMarkerPrefix, minorRelease)
 	return clusterctl.ResolveRelease(ctx, releaseMarker)
 }
+
+// verifyV1Beta2Conditions checks the Cluster and Machines of a Cluster that
+// the given v1beta2 condition types are set to true without a message, if they exist.
+func verifyV1Beta2Conditions(ctx context.Context, c client.Client, clusterName, clusterNamespace string, v1beta2conditionTypes map[string]struct{}) {
```

Comment:
can we split this in two functions:

func verifyClusterAvailabe(ctx context.Context, c client.Client, clusterName, clusterNamespace string) {
```
func verifyV1Beta2Conditions(ctx context.Context, c client.Client, clusterName, clusterNamespace string, v1beta2conditionTypes map[string]struct{}) {
```

```
func verifyMachinesReady(ctx context.Context, c client.Client, clusterName, clusterNamespace string) {
```

Note: I don't think we need to accept a list of conditions in input, those are the two we care about + other conditions might have negative polarity + using a map for this parameter is not ideal